osu.Game/Online/OnlineViewLayer.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Online.Placeholders;
+using osuTK.Graphics;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A layer which displays on top of content which require a local user to be logged in.
+    /// Blocks input to the underlying content, dims the optional target content and displays a login placeholder when the user isn't logged in.
+    /// </summary>
+    public class OnlineViewLayer : VisibilityContainer, IOnlineComponent
+    {
+        protected LoadingSpinner LoadingSpinner;
```

Comment:
If this is exposed just for the test, might as well use `ChildrenOfType<>` for that too instead and not expose.